### PR TITLE
feat(minor): open v2 profile api support collaboration

### PR DESCRIPTION
**Pull request by nannan00**
### Description

Fixes # (issue)

### Checklist

- [ ] 填写 PR 描述及相关 issue (write PR description and related issue)
- [ ] 代码风格检查通过 (code style check passed)
- [ ] PR 中包含单元测试 (include unit test)
- [ ] 单元测试通过 (unit test passed)
- [ ] 本地开发联调环境验证通过 (local development environment verification passed)

### CODE REVIEW

File: src/bk-user/bkuser/apis/open_v2/views/profilers.py [hunk 1]
```diff
@@ -472,14 +551,13 @@ def _build_user_info(self, tenant_user: TenantUser, fields: List[str]) -> Dict[s
...
-            "staff_status": "",
+            "status": TENANT_USER_STATUS_TO_PROFILE_STATUS_MAP.get(tenant_user.status, tenant_user.status),
             "enabled": True,
-            # TODO: 协同时需要调整为按照协同租户配置的用户自定义字段进行输出
+            # 2.x 版本里不同数据源，自定义字段本身就不一致，所以
             "extras": extras,
-            "position": int(extras.get("position", 0)),
+            "position": int(source_extras.get("position", 0)),
             # 总是返回固定值
+            "staff_status": "IN",
```
Comment (narasux) [inline src/bk-user/bkuser/apis/open_v2/views/profilers.py:560]: IN ？

File: src/bk-user/bkuser/apis/open_v2/views/profilers.py [hunk 2]
```diff
@@ -472,14 +551,13 @@ def _build_user_info(self, tenant_user: TenantUser, fields: List[str]) -> Dict[s
             "wx_openid": tenant_user.wx_openid,
             "domain": tenant_user.data_source.domain,
             "category_id": tenant_user.data_source_id,
-            # FIXME (nan) 新版状态映射成老版状态？
-            "status": tenant_user.status,
-            "staff_status": "",
+            "status": TENANT_USER_STATUS_TO_PROFILE_STATUS_MAP.get(tenant_user.status, tenant_user.status),
             "enabled": True,
-            # TODO: 协同时需要调整为按照协同租户配置的用户自定义字段进行输出
+            # 2.x 版本里不同数据源，自定义字段本身就不一致，所以
```
Comment (narasux) [inline src/bk-user/bkuser/apis/open_v2/views/profilers.py:556]: 所以？

File: src/bk-user/bkuser/apis/open_v2/views/profilers.py [hunk 3]
```diff
@@ -34,22 +35,41 @@
...
+    EXPIRED = EnumField("EXPIRED", label="已过期")
+
+
+TENANT_USER_STATUS_TO_PROFILE_STATUS_MAP = {
+    TenantUserStatus.ENABLED.value: ProfileStatusEnum.NORMAL.value,
+    TenantUserStatus.DISABLED.value: ProfileStatusEnum.DISABLED.value,
+    TenantUserStatus.EXPIRED.value: ProfileStatusEnum.EXPIRED.value,
+}
+
+PROFILE_STATUS_TENANT_USER_STATUS_MAP = {v: k for k, v in TENANT_USER_STATUS_TO_PROFILE_STATUS_MAP.items()}
```
Comment (narasux) [inline src/bk-user/bkuser/apis/open_v2/views/profilers.py:61]: Nit: 保持命名格式一致（都要 TO / 都不要）

File: src/bk-user/bkuser/apis/open_v2/views/profilers.py [hunk 4]
```diff
@@ -253,71 +282,109 @@ def _filter_queryset(self, params: Dict[str, Any]) -> QuerySet[TenantUser]:
         if not lookup_field:
```
Comment (narasux) [inline src/bk-user/bkuser/apis/open_v2/views/profilers.py:282]: `queryset = TenantUser.objects.select_related("data_source_user", "data_source")......` ?

因为我看到下面有取 `tenant_user.data_source.domain`

File: src/bk-user/bkuser/apis/open_v2/views/profilers.py (hunk 1 above)
Comment (nannan00) [reply src/bk-user/bkuser/apis/open_v2/views/profilers.py:560]: 是滴，2.x 版本确实是这个值

File: src/bk-user/bkuser/apis/open_v2/views/profilers.py (hunk 2 above)
Comment (nannan00) [reply src/bk-user/bkuser/apis/open_v2/views/profilers.py:556]: 已删除，前面另外说明了

File: src/bk-user/bkuser/apis/open_v2/views/profilers.py (hunk 3 above)
Comment (nannan00) [reply src/bk-user/bkuser/apis/open_v2/views/profilers.py:61]: DONE

File: src/bk-user/bkuser/apis/open_v2/views/profilers.py (hunk 4 above)
Comment (nannan00) [reply src/bk-user/bkuser/apis/open_v2/views/profilers.py:282]: DONE

**Review by narasux — APPROVED** [review]

**Pull request merged**